src/components/DashboardLayout/InstanceViewLayout.tsx

```diff
@@ -22,8 +27,8 @@ export const InstanceViewLayout = () => {
   useEffect(() => {
     if (queryInstanceId && instances && queryInstanceId in instances) {
       setInstanceState(instances[queryInstanceId]);
-      if (!location.pathname.startsWith('/dashboard'))
-        setPathname('/dashboard');
+      if (!location.pathname.startsWith('/dashboard/overview'))
```

Comment:
Saving any of the settings in the setting tab in the "server setting" section, with enter or the on-screen button, will redirect back to /overview for some reason?
This is likely because these settings update `instances`, and thus cause this line to run again.

Reply:
The redirect happens when an instance changes state too